test/fakeintake/aggregator/common.go

```diff
@@ -126,14 +135,27 @@ func getReadCloserForEncoding(payload []byte, encoding string) (rc io.ReadCloser
 
 // GetPayloadsByName return the payloads for the resource name
 func (agg *Aggregator[P]) GetPayloadsByName(name string) []P {
-	return agg.payloadsByName[name]
+	agg.mutex.RLock()
+	defer agg.mutex.RUnlock()
+	payloads := agg.payloadsByName[name]
+	return payloads
 }
 
 // Reset the aggregation
 func (agg *Aggregator[P]) Reset() {
+	agg.mutex.Lock()
+	defer agg.mutex.Unlock()
 	agg.payloadsByName = map[string][]P{}
 }
 
+// Replace payloads
+func (agg *Aggregator[P]) Replace(payloadsByName map[string][]P) {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Does this function need to be exported ? If not you can just use the given map as is (no need to copy)

Reply:
Changed !